torch_frame/utils/concat.py

```diff
@@ -12,31 +12,90 @@
 from torch_frame.typing import TensorData
 
 
-def cat(tf_list: list[TensorFrame], along: str) -> TensorFrame:
+def cat(
+    lst: list[TensorFrame] | list[TensorData],
+    dim: int,
+) -> TensorFrame | TensorData:
+    r"""Concatenates a list of :class:`TensorFrame` or :class:`TensorData`.
+
+    Args:
+        lst (list[TensorFrame] | list[TensorData]): A list of tensor frames to
+            or tensor data to concatenate.
+        dim (int): Dimension to concatenate tensor data.
+
+    Returns:
+        TensorFrame | TensorData: Concatenated output.
+    """
+    if isinstance(lst[0], TensorFrame):
+        return cat_tensor_frame(lst, dim=dim)
+    else:
+        return cat_tensor_data(lst, dim=dim)
+
+
+def cat_tensor_data(td_list: list[TensorData], dim: int) -> TensorData:
```

Comment:
I think we should treat this as private:
```suggestion
def _cat_tensor_data(td_list: list[TensorData], dim: int) -> TensorData:
```